draft-normington-jsonpath-latest.xml

```diff
@@ -397,57 +397,84 @@ array-slice = [ integer ] ws %x3A ws [ integer ]
-
-                     <t>Slice expression parameters <tt>start</tt> and <tt>end</tt> are not directly usable
-                        as slice bounds and must first be normalized. Normalization is defined as:</t>
-
-                     <sourcecode type="pseudocode">
+                     <section numbered="false" toc="exclude">
+                        <name>Informal Introduction</name>
+                        <t>This section is non-normative.</t>
+                        <t>Array indexing is a way of selecting a particular element of an array using a 0-based index.
+                           For example, the expression <tt>[0]</tt> selects the first element of a non-empty array.
+                        </t>
+                        <t>Negative indices index from the end of an array. For example, the expression <tt>[-2]</tt>
+                           selects the last but one element of an array with at least two elements.
+                        </t>
+                        <t>Array slicing is inspired by the behaviour of the <tt>Array.prototype.slice</tt> method
+                           of the JavaScript language as defined by the ECMA-262 standard
+                           <xref target="ECMA-262" format="default">ECMA-262</xref>,
+                           with the addition of the <tt>step</tt> parameter, which is inspired by the Python slice expression.
+                        </t>
+                        <t>The array slice expression <tt>[start:end:step]</tt> selects elements at indices starting at <tt>start</tt>,
+                           incrementing by <tt>step</tt>, and ending with <tt>end</tt> (which is itself excluded). So, for example,
+                           the expression <tt>[1:3]</tt> (where <tt>step</tt> defaults to <tt>1</tt>) selects elements with indices
+                           <tt>1</tt> and <tt>2</tt> (in that order) whereas <tt>[1:5:2]</tt> selects elements with indices
+                           <tt>1</tt> and <tt>3</tt>.
+                        </t>
+                        <t>When <tt>step</tt> is negative, elements are selected in reverse order. Thus, for example, <tt>[5:1:-2]</tt>
+                           selects elements with indices <tt>5</tt> and <tt>3</tt>, in that order and <tt>[::-1]</tt> selects all the
+                           elements of an array in reverse order.
+                        </t>
+                        <t>When <tt>step</tt> is <tt>0</tt>, no elements are selected. This is the one case which differs from the
+                           behaviour of Python, which raises an error in this case.
+                        </t>
+                        <t>The following section specifies the behaviour fully, without depending on JavaScript or Python behaviour.
+                        </t>
+                     </section>
+                     <section numbered="false" toc="exclude">
+                        <name>Normative Definition</name>
```

Comment:
I'm wondering about specifically flagging this section as normative because that might then cast some doubt over whether other sections are normative. Maybe a bland title such as "Details" would be safer?

Reply:
SGTM

Reply:
Ok, I went for "Detailed Semantics".